hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/LazyDefiningSpecificMemberPms.java

```diff
@@ -113,11 +97,26 @@ public void init(@Nonnull Context context) throws Exception {
             }
         }
         if (address == null && !addresses.contains(address)) {
-            throw new JetException("Cluster does not contain the required member: " + memberAddress);
+            throw new JetException("Cluster does not contain the required member");
```

Comment:
Exception message should contain partitionId if we cannot find owner for the partition

Reply:
but is that possible at all (see javadoc for `ExecutionPlanBuilder.getPartitionAssignment`)?

Reply:
now you create `ExpectNothingProcessorSupplier`, but my question remains: is is possible to not find partition in `partitionAssignment`?

Reply:
I will answer myself: it is possible if `KEY_REQUIRED_PARTITIONS` is inconsistent with partitionId determined for `LazyDefiningSpecificMemberPms`. Such case should be treated as internal error, instead of creating only `ExpectNothingProcessorSupplier` instances and getting some obscure error from them later.

Reply:
>  it is possible if KEY_REQUIRED_PARTITIONS ... 

Again, this PMS is not directly related to partition pruning, it is just a useful complement to it. It may work even without declared strategy.

Reply:
but it has to support all intended usages by SQL

Reply:
BTW: fail-fast and defensive programming